framework/resource_set.go

```diff
@@ -0,0 +1,100 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/micrologger/loggermeta"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/giantswarm/operatorkit/framework/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/resourcecanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updateallowedcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updatenecessarycontext"
+)
+
+type ResourceSetConfig struct {
+	// Handles determines if this resource set handles the reconciliation of the
+	// object.
+	Handles   func(obj interface{}) bool
+	InitCtx   func(ctx context.Context, obj interface{}) (context.Context, error)
```

Comment:
I'd like to keep Func suffix, also to add it to Handles. It's always weird to call a variable.

Reply:
I think it is cleaner that way. Appending a `Func` suffix now everywhere does not seem to add real value.

Reply:
Cleaner yes, but not explicit.

Reply:
It's arguable. Let's not spent more time on that.

Reply:
I can add godoc. Good enough for me.